samples/app/src/androidTest/java/com/sourcepoint/app/v6/MainActivityKotlinTest.kt

```diff
@@ -852,6 +852,51 @@ class MainActivityKotlinTest {
         wr { checkPurposesTab() }
     }
 
+    /**
+     * Regression test that verifies if the local data versioning works correctly, meaning that if
+     * the local data version is not the same as the hardcoded one, then the app should call
+     * /consent-status and update the local data version to the hardcoded one
+     */
+    @Test
+    fun given_non_eligible_local_data_version_then_should_call_consent_status_and_update_local_data_version() = runBlocking<Unit> {
+
+        val spClient = mockk<SpClient>(relaxed = true)
+        val initialLocalDataVersion = 0
+
+        /**
+         * This value should be changed each time the HARDCODED_LOCAL_DATA_VERSION value from
+         * DataStorage is updated
+         */
+        val hardcodedLocalDataVersion = 1
+
+        loadKoinModules(
+            mockModule(
+                spConfig = spConf,
+                gdprPmId = "488393",
+                ccpaPmId = "509688",
+                spClientObserver = listOf(spClient),
+                diagnostic = mutableListOf(Pair("sp.key.localDataVersion", 0)),
```

Comment:
> here we want to test the condition where the localDataVersion is different compared with the hardcoded value

The `localDataVersion` value (which is 0) is indeed different compared to the hardcoded value (which is 1).

> You need just to add a GDPR existing consent

I don't get the meaning behind it, since all we want to test is - whenever the `localDataVersion` is different compared to the `HARDCODED_VALUE`, we update the `localDataVersion` to the `HARDCODED_VALUE`

Reply:
Clarified with a call